Sources/TuistHasher/ResourcesContentHasher.swift

```diff
@@ -3,43 +3,78 @@ import TuistCore
 }
 
 /// `ResourcesContentHasher`
 /// is responsible for computing a unique hash that identifies a list of resources
 public final class ResourcesContentHasher: ResourcesContentHashing {
     private let contentHasher: ContentHashing
     private let privacyManifestContentHasher: PrivacyManifestContentHasher
+    private let platformConditionContentHasher: PlatformConditionContentHashing
 
     // MARK: - Init
 
     public convenience init(contentHasher: ContentHashing) {
         self.init(
             contentHasher: contentHasher,
-            privacyManifestContentHasher: PrivacyManifestContentHasher(contentHasher: contentHasher)
+            privacyManifestContentHasher: PrivacyManifestContentHasher(contentHasher: contentHasher),
+            platformConditionContentHasher: PlatformConditionContentHasher(contentHasher: contentHasher)
         )
     }
 
     public init(
         contentHasher: ContentHashing,
-        privacyManifestContentHasher: PrivacyManifestContentHasher
+        privacyManifestContentHasher: PrivacyManifestContentHasher,
+        platformConditionContentHasher: PlatformConditionContentHashing
     ) {
         self.contentHasher = contentHasher
         self.privacyManifestContentHasher = privacyManifestContentHasher
+        self.platformConditionContentHasher = platformConditionContentHasher
     }
 
     // MARK: - ResourcesContentHashing
 
-    public func hash(resources: ResourceFileElements) throws -> String {
-        var hashes = try resources.resources
-            .sorted(by: { $0.path < $1.path })
-            .map { try contentHasher.hash(path: $0.path) }
+    public func hash(identifier: String, resources: ResourceFileElements) throws -> MerkleNode {
+        var children: [MerkleNode] = []
+
+        children
+            .append(
+                contentsOf: try resources.resources.sorted(by: { $0.path < $1.path })
+                    .map { try hashResourceFileElement(element: $0) }
+            )
 
         if let privacyManifest = resources.privacyManifest {
-            hashes.append(try privacyManifestContentHasher.hash(privacyManifest))
+            children.append(try privacyManifestContentHasher.hash(
+                identifier: "privacyManifest",
+                privacyManifest: privacyManifest
+            ))
+        }
+
+        return MerkleNode(
+            hash: try contentHasher.hash(children.map(\.hash)),
+            identifier: identifier,
+            children: children
+        )
+    }
+
+    private func hashResourceFileElement(element: ResourceFileElement) throws -> MerkleNode {
+        var children: [MerkleNode] = []
+
+        children.append(MerkleNode(hash: try contentHasher.hash(path: element.path), identifier: "content"))
+        children.append(MerkleNode(hash: try contentHasher.hash(element.isReference), identifier: "isReference"))
+        children.append(MerkleNode(hash: try contentHasher.hash(element.tags), identifier: "tags"))
```

Comment:
```suggestion
        var children: [MerkleNode] = [
          MerkleNode(hash: try contentHasher.hash(path: element.path), identifier: "content"),
          MerkleNode(hash: try contentHasher.hash(element.isReference), identifier: "isReference"),
          MerkleNode(hash: try contentHasher.hash(element.tags), identifier: "tags")
        ]
```

I'd also try to get rid of some of the boilerplate, so it's easier to read. I'd generally use `children.append` only when a node is added in certain conditions only. The rest that's added always can be added directly in the array.